kasa/device.py

```diff
@@ -0,0 +1,346 @@
+"""Module for Device base class."""
+import logging
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from datetime import datetime
+from typing import Any, Dict, List, Optional, Sequence, Set, Union
+
+from .credentials import Credentials
+from .device_type import DeviceType
+from .deviceconfig import DeviceConfig
+from .emeterstatus import EmeterStatus
+from .exceptions import SmartDeviceException
+from .iotprotocol import IotProtocol
+from .protocol import BaseProtocol
+from .xortransport import XorTransport
+
+
+@dataclass
+class WifiNetwork:
+    """Wifi network container."""
+
+    ssid: str
+    key_type: int
+    # These are available only on softaponboarding
+    cipher_type: Optional[int] = None
+    bssid: Optional[str] = None
+    channel: Optional[int] = None
+    rssi: Optional[int] = None
+
+    # For SMART devices
+    signal_level: Optional[int] = None
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class Device(ABC):
+    """Placeholder for interface or base class."""
```

Comment:
```suggestion
    """Common device interface."""
```

Reply:
I know you mentioned adding docs in a separate PR, here's one place where we should instruct not to use this class directly but do X instead.